base/inference.jl

```diff
@@ -1287,26 +1283,13 @@ function abstract_call_gf_by_type(f::ANY, atype::ANY, sv::InferenceState)
         lsig = length(msig.parameters)
         ls = length(sigtuple.parameters)
         td = type_depth(sig)
-        # look at the existing edges to detect growing argument lists
         mightlimitlength = ls > lsig + 1
         mightlimitdepth = td > 2
-
         limitlength = false
-        if mightlimitlength
-            for (callee, _) in sv.edges
-                callee = callee::InferenceState
-                if method === callee.linfo.def && ls > length(unwrap_unionall(callee.linfo.specTypes).parameters)
-                    limitlength = true
-                    break
-                end
-            end
-        end
-
-        # limit argument type size growth
         if mightlimitlength || mightlimitdepth
             # TODO: FIXME: this heuristic depends on non-local state making type-inference unpredictable
```

Comment:
Unfortunately not, but this PR does enable a code rewrite to fix the issue. This is also relevant to your other question:

> Does this do anything to avoid using a very large amount of stack space?

In short, nope. @vtjnash believes that the future rewrite of the call recursion limit code here will help with stack space usage.